Core/src/org/sleuthkit/autopsy/datamodel/ImageNode.java

```diff
@@ -113,6 +114,9 @@ private void removeListeners() {
         actionsList.add(new RunIngestModulesAction(Collections.<Content>singletonList(content)));
         actionsList.add(new NewWindowViewAction(
                 NbBundle.getMessage(this.getClass(), "ImageNode.getActions.viewInNewWin.text"), this));
+        if (checkSchemaVersion()) {
```

Comment:
Call performed on EDT.

Reply:
Call does not need to be repeated.

Reply:
@dannysmyda, I'm not sure there is a better alternative to making this query in the EDT. Suggestions? Also, which call is repeated?

Reply:
checkSchemaVersion is repeated for all right clicks. I think it should be done once before it is visible.

Reply:
removed Select from imagenode.  Calls to sk public api to get the creation schema version now.

Reply:
Danny's point is that getActions will be called on every ImageNode that is constructed and each query will do a database query in the EDT. The database schema version will not change from call to call to getDBCreationSchemaVersion for a given case. So we should make sure that getDBCreationSchemaVersion saves the result in a private instance variable to avoid the unnecessary database round trips.

Reply:
Never mind, already done!